### PR TITLE
Automate deployment of mvn artifacts

### LINKED ISSUE #674: Automate deployment of maven artifacts

### PR DESCRIPTION
**Pull request by justinharringa**
To fix #674: This work does the following:
* Control deploy via pull requests to the `master` branch. The `deploy` step will only run on that branch and will run the `.travis.deploy.sh` script
    * Add appropriate `.travis.settings.xml` to set maven info
    * version number is controlled by the `project.number` file in the root of the repo. This is where you'd increment your version number (for now, at least)
    * secure creds are encrypted by Travis CI's encrypted credentials functionality

### CODE REVIEW

File: .travis.settings.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        <profile>
+            <id>ossrh</id>
+            <activation>
+                <activeByDefault>true</activeByDefault>
+            </activation>
+            <properties>
+                <gpg.executable>gpg</gpg.executable>
+                <gpg.keyname>${env.GPG_KEY_NAME}</gpg.keyname>
+                <gpg.passphrase>${env.GPG_PASSPHRASE}</gpg.passphrase>
+            </properties>
```
Comment (bsura) [inline .travis.settings.xml]: Where and how the the following 4 env variables set: CI_DEPLOY_USER, CI_DEPLOY_PASSWORD, GPG_KEY_NAME and GPG_PASSPHRASE?

File: .travis.deploy.sh [hunk 2]
```diff
@@ -0,0 +1,7 @@
+#!/usr/bin/env bash
+cp .travis.settings.xml $HOME/.m2/settings.xml
+# PROJECT_VERSION in the commands below should be incremented and sourced from project.version
+source project.version
+# Builds top level pom w/ zip
+mvn --batch-mode --non-recursive -DskipTests -DskipDockerBuild -U -Prelease -Dproject.version=${PROJECT_VERSION} clean install
+mvn --batch-mode -DskipDockerBuild -DskipTests -Dproject.version=${PROJECT_VERSION} -Prelease clean deploy
```
Comment (bsura) [inline .travis.deploy.sh]: Where will this deploy to? Maven central?

File: .travis.settings.xml (hunk 1 above)
Comment (justinharringa) [reply .travis.settings.xml]: Just for those who weren't there when we happened to talk about this in the kitchen: 😄 
These variables are sourced from secrets in .travis.yml that I added as part of the PR.

File: .travis.deploy.sh (hunk 2 above)
Comment (justinharringa) [reply .travis.deploy.sh]: This is all set up via your [pom.xml](https://github.com/salesforce/Argus/blob/7a1038081d70139f4fa10409b58714b784a5a5d6/pom.xml#L314) as well as the `.travis.settings.xml`. The [docs](http://central.sonatype.org/pages/apache-maven.html) describe this. Looks like it should automatically stage to the OSSRH repo and then automatically push to maven central if all of the Nexus checks on OSSRH go well.

**Pull request merged**